integration/query_projection_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestQueryProjectionPositionalOperatorErrors(t *testing.T) {
+	t.Parallel()
+
+	ctx, coll := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
+
+	for name, tc := range map[string]struct { //nolint:vet // used for testing only
```

Comment:
Oh, wow! Great job figuring out these cases! Overall, it feels like most of the position operator cases rather give errors and there is only a particular subset of queries that will work 😅

Reply:
The trickiest one is the case Dmitry found about mismatch, in fact it is documented. It describes undefined behavior though, but we now know it's error https://www.mongodb.com/docs/manual/reference/operator/projection/positional/#array-field-limitations

> Only one array field should appear in the [query document](https://www.mongodb.com/docs/manual/tutorial/query-documents/#std-label-read-operations-query-argument). Additional array fields in the query document may lead to undefined behavior.